### PR TITLE
refactor(notebooks): use internal adam for autograd27 smatrix

### PR DESCRIPTION
**Pull request by marcorudolphflex**
## Summary

This updates `Autograd27Smatrix.ipynb` as a standalone extraction from #465.

- Replaces the direct `optax` optimizer setup and manual update loop with Tidy3D's internal `adam` and `optimize` helpers.
- Uses `learning_rate = 0.04`, which avoids the large objective spike seen with the earlier `0.05` rerun.
- Removes the notebook-level `optax` dependency.
- Sanitizes local output metadata and local path warning outputs from the rerun.

## Optimization Result

Lower-LR internal-Adam rerun, final saved step:

- Step: `25`
- Objective `J`: `2.6270e-02` (vs 1.0909e-02 before)
- Gradient norm: `1.5076e-02`
before 
image

after
image

## Validation

- `uvx ruff format --check --diff Autograd27Smatrix.ipynb`
- `uvx ruff check Autograd27Smatrix.ipynb`
- `uv run --no-project python scripts/sync_metadata_tags.py --check`
- `uv run --no-project python scripts/validate_notebook_metadata.py Autograd27Smatrix.ipynb`
- `uv run --no-project python scripts/check_notebook_private_paths.py Autograd27Smatrix.ipynb`
- `python3 misc/check_misc_references.py`
- `uv run spellcheck.py Autograd27Smatrix.ipynb`

---

> [!NOTE]
> **Low Risk**
> Notebook-only refactor with no production code paths; main review note is that the rerun yields a higher final objective than before, which is expected from the hyperparameter change rather than a functional regression.
> 
> **Overview**
> **Autograd27Smatrix.ipynb** now follows the same optimization pattern as other autograd examples: the inverse-design loop uses **`tidy3d.plugins.autograd`** **`adam`** and **`optimize`** instead of a third-party **`optax`** setup and hand-written update steps.
> 
> The optimization cell wires **`record_step`** as a **`optimize`** callback (density plots plus step logs for **J**, **beta**, and gradient norm) and keeps parameters in **[0, 1]** via **`bounds`**. **`learning_rate`** is set to **`0.04`** for the rerun (replacing **`0.05`**), which the author reports avoids a large objective spike. Committed outputs were refreshed for the new run (e.g. final step 25, **J ≈ 2.63e-02**).
> 
> This is documentation/example-only: no library or simulation API changes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fa22e8cb03de6e5504710338910fd685037c8ec4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by groberts-flex — APPROVED** [review]
looks good thanks @marcorudolphflex
